### PR TITLE
add e2e test for mysql plugin

### PR DESCRIPTION
**Pull request by jockech**
Create the e2e test for mysql plugin.

snapshot:
image

result:
image

### CODE REVIEW

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    // Insert dummy data.
+    statement.executeUpdate("INSERT INTO " + sourceTable + " (id, lastName)" +
+                       "VALUES (1, 'Simpson')");
+    statement.executeUpdate("INSERT INTO " + sourceTable + " (id, lastName)" +
+                       "VALUES (2, 'McBeal')");
+    statement.executeUpdate("INSERT INTO " + sourceTable + " (id, lastName)" +
+                       "VALUES (3, 'Flinstone')");
+
+
+    String createTargetTableQuery = "CREATE TABLE IF NOT EXISTS " + targetTable +
```
Comment (itsankit-google) [inline mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java]: Can we separate out setting up of `targetTable` in another method?
If `sourceTable` and `targetTable` are same we might end up with unexpected results.

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    statement.executeUpdate(createTargetTableQuery);
+
+    // Truncate table to clean the data of last failure run.
+    String truncateTargetTableQuery = "TRUNCATE TABLE " + targetTable;
+    statement.executeUpdate(truncateTargetTableQuery);
+    statement.close();
+    connect.close();
+  }
+
+  public static void dropTables(String sourceTable, String targetTable) throws SQLException, ClassNotFoundException {
```
Comment (itsankit-google) [inline mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java]: Can we pass the input parameter as list of tables to drop instead of individual tables?

File: pom.xml [hunk 3]
```diff
@@ -62,7 +62,7 @@
     <!-- version properties -->
     <cdap.version>6.8.0-SNAPSHOT</cdap.version>
     <cdap.plugin.version>2.10.0-SNAPSHOT</cdap.plugin.version>
-    <guava.version>13.0.1</guava.version>
+    <guava.version>27.0.1-jre</guava.version>
```
Comment (itsankit-google) [inline pom.xml]: why is this change needed?

**Review by itsankit-google — COMMENTED** [review]
Few comments,

Are we expecting jdbc drivers to be already installed?

File: pom.xml (hunk 3 above)
Comment (jockech) [reply pom.xml]: I have got `java.lang.NoSuchMethodError: 'java.util.stream.Collectorcom.google.common.collect.ImmutableList.toImmutableList()'`
and checked the [stackoverflow](https://stackoverflow.com/questions/65973443/java-lang-nosuchmethoderror-java-util-stream-collector-com-google-common-colle) and decided to syncup with [goolge-cloud](https://github.com/data-integrations/google-cloud/blob/de29c557c38a2322c807dbdf7061342c72b333f7/pom.xml#L90)

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ *  MySQL connector
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(MysqlConnector.NAME)
+@Category("MySQL")
+@Description("Connection to access data in MySQL.")
```
Comment (itsankit-google) [inline mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java]: Why are these annotations needed? it is not a CDAP Connector plugin.

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java (hunk 2 above)
Comment (jockech) [reply mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java]: > Few comments,
> 
> Are we expecting jdbc drivers to be already installed?

Yes. Don't think we could install the driver in the test.

File: mysql-plugin/src/e2e-test/resources/errorMessage.properties [hunk 5]
```diff
@@ -0,0 +1 @@
+validationSuccessMessage=No errors found.
```
Comment (rmstar) [inline mysql-plugin/src/e2e-test/resources/errorMessage.properties:1]: where is this used?

**Review by rmstar — COMMENTED** [review]
Do we need [pluginDataCyAttributes.properties](https://github.com/data-integrations/database-plugins/pull/273/files#diff-078674b9e17146b5b1fef0e32bde49c6d88c12943af69ee3f1d64ef051e03fab). Looks like it's empty?

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ *  MySQL connector
+ */
+public class MysqlConnector {
```
Comment (itsankit-google) [inline mysql-plugin/src/e2e-test/java/io/cdap/plugin/MysqlConnector.java]: Can we rename it to `MysqlClient` or `MysqlHelper`? 
Want to avoid conflict with CDAP connector terminology.

**Review by itsankit-google — APPROVED** [review]
One minor comment, otherwise LGTM

File: mysql-plugin/src/e2e-test/resources/pluginParameters.properties [hunk 7]
```diff
@@ -0,0 +1,12 @@
+driver=mysql
+host=34.105.111.83
+username=dummy
+password=dummy
+database=sakila
+port=3306
+selectQuery=select * from
```
Comment (itsankit-google) [inline mysql-plugin/src/e2e-test/resources/pluginParameters.properties]: nit: `selectQuery=select * from ${table}` looks more readable.
Can replace `${table}` with `sourceTable` in test setup hook.

**Review by itsankit-google — APPROVED** [review]
